CorsixTH/Lua/entities/humanoids/vip.lua

```diff
@@ -126,62 +181,85 @@ function Vip:goHome()
   self:despawn()
 end
 
+function Vip:evaluateRoom()
+  local room_extinguisher = 0
+  local room_plant = 0
+  local room_bin = 0
+  -- Another room visited.
+  self.num_visited_rooms = self.num_visited_rooms + 1
+  local room = self.next_room
+  -- if the player is about to kill a live patient for research, punish hard
+  if room.room_info.id == "research" then
+    if room:getPatient() then
+      self.vip_rating = self.vip_rating + 6
+    end
+  end
+  -- evaluate the room we're currently looking at
+  for object, _ in pairs(room.objects) do
+    if object.object_type.id == "extinguisher" and room_extinguisher == 0 then
+      self.room_eval = self.room_eval + 1
+      -- only count this object type once
+      room_extinguisher = 1
+    elseif object.object_type.id == "plant" then
+      room_plant = object:isDying() and room_plant - 1 or room_plant + 1
+    elseif object.object_type.id == "bin" and room_bin == 0 then
+      self.room_eval = self.room_eval + 1
+      -- only count this object type once
+      room_bin = 1
+    end
+
+    if object.strength then
```

Comment:
Should we use object.object_type.id == "machine" here instead of checking for a strength attribute considering we've abstracted from this in the evaluation?

Reply:
In the current code we don't have a flag for an object being a machine except by its "strength". Not sure if we want to tag all machines with an identifier though as a machine.